pkg/cmd/admin/migrate/migrator.go

```diff
@@ -0,0 +1,362 @@
+package migrate
+
+import (
+	"fmt"
+	"io"
+	"strings"
+
+	"github.com/golang/glog"
+	"github.com/spf13/cobra"
+
+	"k8s.io/kubernetes/pkg/api/errors"
+	"k8s.io/kubernetes/pkg/kubectl"
+	"k8s.io/kubernetes/pkg/kubectl/resource"
+	"k8s.io/kubernetes/pkg/util/sets"
+
+	cmdutil "github.com/openshift/origin/pkg/cmd/util"
+	"github.com/openshift/origin/pkg/cmd/util/clientcmd"
+)
+
+// MigrateVisitFunc is invoked for each returned object, and may return a
+// Reporter that can contain info to be used by save.
+type MigrateVisitFunc func(info *resource.Info) (Reporter, error)
+
+// MigrateActionFunc is expected to persist the altered info.Object. The
+// Reporter returned from Visit is passed to this function and may be used
+// to carry additional information about what to save on an object.
+type MigrateActionFunc func(info *resource.Info, reporter Reporter) error
+
+// Reporter indicates whether a resource requires migration.
+type Reporter interface {
+	// Changed returns true if the resource requires migration.
+	Changed() bool
+}
+
+// ResourceOptions assists in performing migrations on any object that
+// can be retrieved via the API.
+type ResourceOptions struct {
+	In          io.Reader
+	Out, ErrOut io.Writer
+
+	Include       []string
+	Filenames     []string
+	AllNamespaces bool
+	Confirm       bool
+	Output        string
+
+	DefaultExcludes []string
+
+	Builder   *resource.Builder
+	SaveFn    MigrateActionFunc
+	PrintFn   MigrateActionFunc
+	DryRun    bool
+	Summarize bool
+}
+
+func (o *ResourceOptions) Bind(c *cobra.Command) {
+	c.Flags().StringVarP(&o.Output, "output", "o", o.Output, "Output the modified objects instead of saving them, valid values are 'yaml' or 'json'")
+	c.Flags().StringSliceVar(&o.Include, "include", o.Include, "Resource types to migrate. Passing --filename will override this flag.")
+	c.Flags().BoolVar(&o.Confirm, "confirm", false, "If false no migrations will be performed.")
+	c.Flags().BoolVar(&o.AllNamespaces, "all-namespaces", true, "If true, all namespaces will be utilized.")
```

Comment:
That's why I don't see this option being useful at all. Either you apply to all (by default) or per namespace (if specified).